pyamg/util/utils.py

```diff
@@ -2044,13 +2051,23 @@ def filter_matrix_rows(A, theta):
     if (theta < 0) or (theta >= 1.0):
         raise ValueError('theta must be in [0,1)')
 
+    # Filter by diagonal entry, A_ij = 0 if |A_ij| < theta*|A_ii|
+    if diagonal:
+        amg_core.filter_matrix_rows(A.shape[0], theta, A.indptr,
+                                    A.indices, A.data, lump)
+        A.eliminate_zeros()
+        if Aformat == 'bsr':
+            A = A.tobsr(blocksize=blocksize)
```

Comment:
Just a code style thing, below in line line 286 we say A_filter.tobsr(blocksize).  Maybe to match line 2060, 286 should be A_filter.tobsr(blocksize=blocksize) ... ?  But, this isn't a big deal